src/controllers/controller.h

```diff
@@ -193,13 +196,33 @@ class ControllerJSProxy : public QObject {
             : m_pController(m_pController) {
     }
 
+    QHash<QString, QString> m_cache;
+
     // The length parameter is here for backwards compatibility for when scripts
     // were required to specify it.
     Q_INVOKABLE virtual void send(const QList<int>& data, unsigned int length = 0) {
         Q_UNUSED(length);
         m_pController->send(data, data.length());
     }
 
+    // Available charsets should be available here:
+    // http://www.iana.org/assignments/character-sets/character-sets.xhtml
+    Q_INVOKABLE QByteArray convertEncoding(const QString& targetCharset, const QString& value) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+        auto* codec = QTextCodec::codecForName(targetCharset.toUtf8());
+        if (!codec) {
+            return nullptr;
+        }
+        return codec->fromUnicode(value);
+#else
+        QStringEncoder fromUtf8 = QStringEncoder(targetCharset.toUtf8().data());
+        if (!fromUtf8.isValid()) {
+            return nullptr;
+        }
+        return fromUtf8(value);
```

Comment:
maybe an explicit cast to QByteArray will fix it?
```suggestion
        QStringEncoder fromUtf16 = QStringEncoder(targetCharset.toUtf8().data());
        if (!fromUtf16.isValid()) {
            return nullptr;
        }
        return QByteArray(fromUtf16(value));
```

Reply:
`DecodedData`  already has a `operator QByteArray()` and [that's what's documented](https://doc.qt.io/qt-6/qstringencoder.html#details).

Reply:
No that was me. It correcty returns an `ArrayBuffer`. I just didn't know how to use it.